bindings/pyroot/pythonizations/python/ROOT/__init__.py

```diff
@@ -60,7 +60,70 @@
 import sys
 from ._facade import ROOTFacade
 
-sys.modules[__name__] = ROOTFacade(sys.modules[__name__], _is_ipython)
+_root_facade = ROOTFacade(sys.modules[__name__], _is_ipython)
+sys.modules[__name__] = _root_facade
+
+# Configure meta-path finder for ROOT namespaces, following the Python
+# documentation and an example:
+#
+#   * https://docs.python.org/3/library/importlib.html#module-importlib.abc
+#
+#   * https://python.plainenglish.io/metapathfinders-or-how-to-change-python-import-behavior-a1cf3b5a13ec
+from importlib.abc import Loader, MetaPathFinder
+from importlib.machinery import ModuleSpec
+from importlib.util import spec_from_loader
+
+
+def _lookup_root_module(fullname: str):
+    """
+    Recursively looks up attributes of the ROOT facade
+    (usually C++ namespaces), using a full module name.
+    """
+    keys = fullname.split(".")[1:]
+    ret = _root_facade
+    for part in keys:
+        ret = getattr(ret, part, None)
+        if ret is None:
+            return None
+    return ret
+
+
+class _RootNamespaceLoader(Loader):
+    """
+    Custom loader for modules under the ROOT namespace.
+    """
+
+    def is_package(self, fullname: str) -> bool:
+        """
+        Indicates whether the given module is considered a package.
+        We pretend to Python that everything in ROOT can be used like a package.
+        """
```

Comment:
Why? So that we can chain imports like in `from ROOT.RDF.Experimental....` ?

Reply:
Maybe give a bit more explanation in the docstring

Reply:
I now explicitly check if the attribute can actually be considered a submodule, so I don't need to explain anything anymore :) (I hope :D)